janus.c

```diff
@@ -3059,6 +3062,7 @@ void janus_plugin_end_session(janus_plugin_session *plugin_session) {
 	if((plugin_session < (janus_plugin_session *)0x1000) || !janus_plugin_session_is_alive(plugin_session) || g_atomic_int_get(&plugin_session->stopped))
 		return;
 	GSource *timeout_source = g_timeout_source_new_seconds(0);
+	janus_refcount_increase(&plugin_session->ref);
```

Comment:
Can you move the increase before the `g_timeout_source_new_seconds`? Won't buy much, but will at least immediately follow the checks above.